### PR TITLE
Update builder code to Capella

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Implements https://github.com/ethereum/builder-specs/pull/60

**Which issues(s) does this PR fix?**

Part of #11077

### CODE REVIEW

File: api/client/builder/client.go [hunk 1]
```diff
@@ -247,12 +276,78 @@ func (c *Client) RegisterValidator(ctx context.Context, svr []*ethpb.SignedValid
...
+		}
+		return blocks.WrappedExecutionPayloadCapella(p)
+	default:
+		return nil, fmt.Errorf("unsupported block version %s", version.String(sb.Version()))
+	}
+}
+
+// SubmitBlindedBlockCapella calls the builder API endpoint that binds the validator to the builder and submits the block.
+// The response is the full ExecutionPayloadCapella used to create the blinded block.
+func (c *Client) SubmitBlindedBlockCapella(ctx context.Context, sb *ethpb.SignedBlindedBeaconBlockCapella) (*v1.ExecutionPayloadCapella, error) {
```
Comment (terencechain) [inline api/client/builder/client.go:346]: This will be v2 path

File: api/client/builder/client.go [hunk 2]
```diff
@@ -202,8 +207,8 @@ func execHeaderPath(slot types.Slot, parentHash [32]byte, pubkey [48]byte) (stri
 	return b.String(), nil
 }
 
-// GetHeader is used by a proposing validator to request an ExecutionPayloadHeader from the Builder node.
-func (c *Client) GetHeader(ctx context.Context, slot types.Slot, parentHash [32]byte, pubkey [48]byte) (*ethpb.SignedBuilderBid, error) {
+// GetHeader is used by a proposing validator to request an execution payload header from the Builder node.
+func (c *Client) GetHeader(ctx context.Context, slot types.Slot, parentHash [32]byte, pubkey [48]byte) (SignedBid, error) {
 	path, err := execHeaderPath(slot, parentHash, pubkey)
```
Comment (terencechain) [inline api/client/builder/client.go:212]: This will be V2 path as well

File: api/client/builder/client.go (hunk 1 above)
Comment (rkapka) [reply api/client/builder/client.go:346]: Talked to Alex, the existing one supports Capella

File: api/client/builder/client.go (hunk 2 above)
Comment (rkapka) [reply api/client/builder/client.go:212]: Talked to Alex, the existing one supports Capella

**Review by terencechain — APPROVED** [review]

**Pull request merged**